src/lib.rs

```diff
@@ -12,38 +12,36 @@
 
 #[macro_use]
 extern crate error_chain;
-#[macro_use]
-extern crate lazy_static;
 
 pub mod result;
 
 use result::{ErrorKind, Result};
 
 pub(crate) const MAXIMUM_K_SIZE: usize = u32::max_value() as usize;
 
-// Note: Replace this with a const fn when it's available in stable
-// https://github.com/rust-lang/rust/issues/24111
-lazy_static! {
-    static ref H_LOOKUP: [u64; 256] = {
-        let mut lookup = [1; 256];
-        lookup[b'A' as usize] = 0x3c8b_fbb3_95c6_0474;
-        lookup[b'C' as usize] = 0x3193_c185_62a0_2b4c;
-        lookup[b'G' as usize] = 0x2032_3ed0_8257_2324;
-        lookup[b'T' as usize] = 0x2955_49f5_4be2_4456;
-        lookup[b'N' as usize] = 0;
-        lookup
-    };
-    static ref RC_LOOKUP: [u64; 256] = {
-        let mut lookup = [1; 256];
-        lookup[b'A' as usize] = 0x2955_49f5_4be2_4456;
-        lookup[b'C' as usize] = 0x2032_3ed0_8257_2324;
-        lookup[b'G' as usize] = 0x3193_c185_62a0_2b4c;
-        lookup[b'T' as usize] = 0x3c8b_fbb3_95c6_0474;
-        lookup[b'N' as usize] = 0;
-        lookup
-    };
+const fn h_lookup_table() -> [u64; 256] {
+    let mut lookup = [1; 256];
```

Comment:
Just out of curiosity, any reason to initialize it with 256 items instead of just 5?

Reply:
We are not converting the bases `ACGT` to a 2-bit representation, instead we keep it as `char`, which is `u8` (and 2**8 = 256).

That said, we could return `[u8; 256]` instead, because the two places using the lookup table are doing a `c as usize` conversion:
https://github.com/luizirber/nthash/pull/6/files#diff-b4aea3e418ccdb71239b96952d9cddb6R47
https://github.com/luizirber/nthash/pull/6/files#diff-b4aea3e418ccdb71239b96952d9cddb6R56